drivers/soundwire/intel.c

```diff
@@ -846,6 +849,16 @@ static int intel_hw_params(struct snd_pcm_substream *substream,
 	if (!dma)
 		return -EIO;
 
+	if (dma->hw_params) {
```

Comment:
I have to  revert "ASoC: SOF: handle paused streams during system suspend" now. Otherwise I will see
```
=== PAUSE ===                                                             Suspended. Trying resume. Failed. Restarting stream. aplay: suspend:1691: suspend: prepare error: Invalid argument
```
```
[  880.519173] sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx: 0x30100000: GLB_TPLG_MSG: PIPE_NEW
[  880.519273] sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx error for 0x30100000 (msg/reply size: 48/20): -22
[  880.519275] sof-audio-pci-intel-tgl 0000:00:1f.3: error: failed to load widget PIPELINE.31.ALH258.OUT
[  880.519276] sof-audio-pci-intel-tgl 0000:00:1f.3: error: failed to restore pipeline after resume -22
[  880.519277] PM: dpm_run_callback(): pci_pm_resume+0x0/0x80 returns -22
[  880.519284] sof-audio-pci-intel-tgl 0000:00:1f.3: PM: failed to resume async: error -22
```
[dmesg.txt](https://github.com/thesofproject/linux/files/7075463/dmesg.txt)

Reply:
Thanks @bardliao there must be something we missed, IIRC you are using firmware 1.8 and we're doing something different. Gah.